File: LeetCode/Two-Sum.cpp
1class Solution {
2public:
3    vector<int> twoSum(vector<int>& nums, int target) {
4        map<int, int> mp;
5        int secNum = 0;
6        for(int i=0; i<nums.size(); i++){
7            secNum = target - nums[i];
8            if(mp.count(secNum)) return {i, mp[secNum]};
9            mp[nums[i]] = i;
10        }
11        return {-1, -1};
12    }
13};

File: LeetCode/Two-Sum.py
1class Solution:
2    def twoSum(self, nums: List[int], target: int) -> List[int]:
3        dic = {}
4        for i in range(0, len(nums), 1):
5            second_num = target - nums[i]
6            if(dic.get(second_num, -1) != -1):
7                return [i, dic[second_num]]
8            dic[nums[i]] = i
9        return [0, 0]